src/Raven.Server/Documents/Subscriptions/LiveSubscriptionPerformanceCollector.cs

```diff
@@ -0,0 +1,305 @@
+                using (context.OpenReadTransaction())
+                {
+                    var inProgressConnection = Database.SubscriptionStorage.GetSubscriptionConnection(context, subscriptionName);
+                    var inProgressBatchStats = inProgressConnection?.Connection?.GetBatchPerformanceStats();
+                    
+                    if (inProgressBatchStats != null &&
+                        inProgressBatchStats.Completed == false &&
+                        inProgressBatchStats.ToBatchPerformanceLiveStatsWithDetails().DocumentsCount > 0 &&
+                        batchAggregators.Contains(inProgressBatchStats) == false)
+                    {
+                        batchAggregators.Add(inProgressBatchStats);
+                    }
+                }
+                
+                // 3. add to results
+                batchPerformance.AddRange(batchAggregators.Select(x => x.ToBatchPerformanceLiveStatsWithDetails()));
+                
+                var subscriptionItem = preparedStats.Find(x => x.TaskName == kvp.Key);
+                if (subscriptionItem != null)
+                {
+                    subscriptionItem.BatchPerformance = batchPerformance.ToArray();
+                }
+            }
+
+            return preparedStats;
+        }
+        
+        private class SubscriptionAndPerformanceConnectionStatsList
+            : HandlerAndPerformanceStatsList<SubscriptionStorage.SubscriptionGeneralDataAndStats, SubscriptionConnectionStatsAggregator>
+        {
+            public SubscriptionAndPerformanceConnectionStatsList(SubscriptionStorage.SubscriptionGeneralDataAndStats subscription) : base(subscription)
+            {
+            }
+        }
+        
+        private class SubscriptionAndPerformanceBatchStatsList
+            : HandlerAndPerformanceStatsList<SubscriptionStorage.SubscriptionGeneralDataAndStats, SubscriptionBatchStatsAggregator>
+        {
+            public SubscriptionAndPerformanceBatchStatsList(SubscriptionStorage.SubscriptionGeneralDataAndStats subscription) : base(subscription)
+            {
+            }
+        }
+        
+        private void OnEndConnection(SubscriptionConnection connection)
+        {
+            var subscriptionName = connection.SubscriptionState.SubscriptionName;
+
+            if (_perSubscriptionConnectionStats.TryGetValue(subscriptionName, out var subscriptionAndStats))
+            {
+                var aggregatorToAdd = connection.GetPerformanceStats();
+                subscriptionAndStats.Performance.TryAdd(aggregatorToAdd);
+            }
+        }
+        
+        private void OnSubscriptionTaskAdded(string subscriptionName)
+        {
+            if (_perSubscriptionConnectionStats.ContainsKey(subscriptionName))
+                return;
+
+            using (Database.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
```

Comment:
why we use server store context pool here, insteaf of db context?

Reply:
Because below `Database.SubscriptionStorage.GetSubscription()` call requires to read data from server store - so it needs to open transaction for that store.